sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -374,6 +375,17 @@ case class InsertIntoTable(
   }
 }
 
+case class InsertIntoDir(
```

Comment:
I could not figure out a way to push the code down without adding some extra if else statements which treats insert statements specially. If you insist I can add the code but I think its better the way it is right now.